gemhardware/include/gem/hw/GEMHwDevice.h

```diff
@@ -123,237 +130,378 @@ namespace gem {
         void reset() { BadHeader=0; ReadError=0; Timeout=0; ControlHubErr=0; return; };
       } DeviceErrors;
 
+      // TODO: REDESIGN
       typedef std::pair<uint8_t, OpticalLinkStatus>  linkStatus;
       //typedef std::vector<linkStatus>                linkStatus;
 
       /**
-       * GEMHwDevice constructor
-       * @param deviceName string to put into the logger
+       * @brief GEMHwDevice constructor based on hostname and connection file
```

Comment:
What happens if the host can't be resolved or reached? Same question for other constructors

Reply:
I believe that for both `xhal` and `uhal` devices, the corresponding construction call will `throw`, so this constructor will not complete